tools/ea/Ea.hs

```diff
@@ -71,50 +67,55 @@ import Pact.Types.SPV (noSPVSupport)
 
 ---
 
-data Env = Env [FilePath]
-
-pEnv :: Parser Env
-pEnv = Env <$> many pTrans
-
-pTrans :: Parser FilePath
-pTrans = strOption
-    (long "transaction" <> metavar "PATH"
-    <> help "Path to YAML file containing a Pact transaction. Can be used multiple times. By default, loads the Coin Contract and Grants files.")
-
 main :: IO ()
 main = do
-    Env txs0 <- execParser opts
-    for_  graphs $ \(v, tag, grants, ns) -> do
-        let txs = bool txs0 [defCoinContract, grants, ns] $ null txs0
-        putStrLn $ "Generating Genesis Payload for " <> show v <> "..."
-        genPayloadModule v tag txs
+    for_ chain0 $ \(v, tag, txs) -> do
+        printf "Generating special Genesis Payload for %s on Chain 0...\n" $ show v
+        genPayloadModule v (tag <> "0") txs
+    for_ otherChains $ \(v, tag, txs) -> do
+        printf "Generating Genesis Payload for %s on all other Chains...\n" $ show v
+        genPayloadModule v (tag <> "N") txs
     putStrLn "Done."
   where
-    opts = info (pEnv <**> helper)
-        (fullDesc <> header "ea - Generate Pact Payload modules")
+    otherChains =
+      [ (Development, "Development", [coinContract, devNGrants, devNs])
+      , (FastTimedCPM petersonChainGraph, "FastTimedCPM", [coinContract, devNGrants, devNs])
+      , (Testnet02, "Testnet", [ coinContract, prodNGrants, prodNs ])
+      ]
 
-    graphs =
-      [ (Development, "Development", devGrants, devNs)
-      , (FastTimedCPM petersonChainGraph, "FastTimedCPM", devGrants, devNs)
-      , (Testnet02, "Testnet", prodGrants, prodNs)
+    chain0 =
+      [ (Development, "Development", [coinContract, devNs, devAllocations, dev0Grants])
+      , (FastTimedCPM petersonChainGraph, "FastTimedCPM", [coinContract, devAllocations, dev0Grants, devNs])
```

Comment:
Perhaps move `devNs` here to the same position that it's in in the other lists.

Reply:
👍